manifests/record/ptr/by_ip.pp

```diff
@@ -0,0 +1,86 @@
+# == Definition: dns::record::ptr::by_ip
+#
+# This type is defined to allow the dns::record::a resource type to
+# use pre-4.x 'iteration' to create reverse ptr records for hosts with
+# multiple ip addresses.
+#
+# === Parameters
+#
+# * `$ip`
+# The ip address for the ptr record.  Defaults to the resource title.
+#
+# * `$host`
+# The (unqualified) host pointed to by this ptr record.
+#
+# * `$ttl`
+# The TTL of the records to be created.  Defaults to undefined.
+#
+# * `$zone`
+# The domain of the host pointed to by this ptr record.
+#
+# === Actions
+#
+# Reformats the $ip, $host, $zone parameters to create a `dns::record::ptr` 
+# resource
+#
+# === Examples
+#
+# @example
+#     dns::record::ptr::by_ip { '192.168.128.42':
+#       host => 'server' ,
+#       zone => 'example.com' ,
+#     }
+#
+# turns into:
+#
+# @example
+#     dns::record::ptr { '42.128.168.192.IN-ADDR.ARPA':
+#       host => '42' ,
+#       zone => '128.168.192.IN-ADDR.ARPA' ,
+#       data => 'server.example.com' ,
+#     }
+#
+# ---
+# @example
+#     dns::record::ptr::by_ip { [ '192.168.128.68', '192.168.128.69', '192.168.128.70' ]:
+#       host => 'multihomed-server' ,
+#       zone => 'example.com' ,
+#     }
+#
+# turns into:
+#
+# @example
+#     dns::record::ptr { '68.128.168.192.IN-ADDR.ARPA':
+#       host => '68' ,
+#       zone => '128.168.192.IN-ADDR.ARPA' ,
+#       data => 'multihomed-server.example.com' ,
+#
+# @example
+#     dns::record::ptr { '69.128.168.192.IN-ADDR.ARPA':
+#       host => '69' ,
+#       zone => '128.168.192.IN-ADDR.ARPA' ,
+#       data => 'multihomed-server.example.com' ,
+#
+# @example
+#     dns::record::ptr { '70.128.168.192.IN-ADDR.ARPA':
+#       host => '70' ,
+#       zone => '128.168.192.IN-ADDR.ARPA' ,
+#       data => 'multihomed-server.example.com' ,
+#     }
+
+define dns::record::ptr::by_ip (
+  $host,
+  $zone,
+  $ttl = undef ,
+  $ip = $name ) {
+
+  $reverse_zone = inline_template('<%= @ip.split(".")[0..-2].reverse.join(".") %>.IN-ADDR.ARPA')
```

Comment:
This is how IN-ADDR.ARPA PTR records work.  It doesn't matter what the subnet size is, the reverse name for IPv4 address a.b.c.d is always found by looking up d.c.b.a.IN-ADDR.ARPA.